### PR TITLE
fix(sign): improve deb signing process

### PR DESCRIPTION
**Pull request by jschmid1**
Signed-off-by: Joshua Schmid <jaiks@posteo.de>

* detached keys are not strictly necessary
* [avoid adding multiple testing keys (the deb signing script will choke on that ](https://github.com/pulp/pulp_deb/blob/289d230a9050f0ea4ae4f7788f011b05e2748f4c/pulp_deb/app/models/signing_service.py#L76-L78)
* spelling
* pathing of files

### CODE REVIEW

**Comment by jschmid1** [comment]
`pulp-core/Dockerfile:36 DL3059 info: Multiple consecutive `RUN` instructions. Consider consolidation.`

that's new :/

**Comment by jschmid1** [comment]
wdyt @dannysauer about disabling DL3059: Multiple consecutive RUN instructions. Consider consolidation.

https://github.com/Kong/docker-pulp/pull/315/commits/5bc2d57b502373befa940cf550606bdaf71802d5

File: pulp-worker/entrypoint.sh [hunk 1]
```diff
@@ -18,6 +19,7 @@ Name-Real: Tester
 Name-Email: test@test.com
 Expire-Date: 0
 EOF
+fi
```
Comment (dannysauer) [inline pulp-worker/entrypoint.sh:22]: This outdented `if` bugs me more than it should. :) 

Then again, so does the heredoc. One way I might work around that is with printf. As in:

```bash
printf "%s\n" \
"Key: val" \
"other: thing" \
| gpg ...
```
Then just the extra quotes would bug me. ;)

**Comment by dannysauer** [comment]
> wdyt @dannysauer about disabling DL3059: Multiple consecutive RUN instructions. Consider consolidation.
> 
> [5bc2d57](https://github.com/Kong/docker-pulp/commit/5bc2d57b502373befa940cf550606bdaf71802d5)

In general I like the reminder, but ultimately I think people editing this dockerfile are likely to make good choices. It's a [slightly contentious check](https://github.com/hadolint/hadolint/issues/715) anyway, so I'm good with that.

**Review by dannysauer — APPROVED** [review]
LGTM

**Pull request merged**